linera-core/src/client/requests_scheduler/scheduler.rs

```diff
@@ -710,70 +707,151 @@ impl<Env: Environment> RequestsScheduler<Env> {
     /// The first successful result, or the last error if all fail
     async fn try_staggered_parallel<T, F, Fut>(
         &self,
         key: &RequestKey,
-        peers: Vec<RemoteNode<Env::ValidatorNode>>,
+        first_peer: RemoteNode<Env::ValidatorNode>,
         operation: &F,
-        staggered_delay_ms: u64,
+        staggered_delay: Duration,
     ) -> Result<T, NodeError>
     where
-        T: Send,
+        T: 'static,
         F: Fn(RemoteNode<Env::ValidatorNode>) -> Fut,
-        Fut: Future<Output = Result<T, NodeError>>,
+        Fut: MaybeSendFuture<Output = Result<T, NodeError>> + 'static,
     {
-        use futures::stream::{FuturesUnordered, StreamExt};
-        use linera_base::time::{timer::sleep, Duration};
+        use futures::{
+            future::{select, Either},
+            stream::{FuturesUnordered, StreamExt},
+            FutureExt,
+        };
+        use linera_base::time::timer::sleep;
+
+        // Use LocalBoxFuture for wasm32, BoxFuture for other targets (which requires Send)
+        #[cfg(target_arch = "wasm32")]
+        type IndexedFuture<T> = futures::future::LocalBoxFuture<'static, Result<T, NodeError>>;
+        #[cfg(not(target_arch = "wasm32"))]
+        type IndexedFuture<T> = futures::future::BoxFuture<'static, Result<T, NodeError>>;
+
+        let mut futures: FuturesUnordered<IndexedFuture<T>> = FuturesUnordered::new();
+        let peer_index = AtomicU32::new(0);
+
+        let push_future = |futures: &mut FuturesUnordered<IndexedFuture<T>>, fut: Fut| {
+            #[cfg(target_arch = "wasm32")]
+            #[allow(clippy::redundant_async_block)]
+            futures.push(async move { fut.await }.boxed_local());
+            #[cfg(not(target_arch = "wasm32"))]
+            // if removed types don't match.
+            #[allow(clippy::redundant_async_block)]
+            futures.push(async move { fut.await }.boxed());
+            peer_index.fetch_add(1, Ordering::SeqCst)
+        };
 
-        let mut futures = FuturesUnordered::new();
+        // Start the first peer immediately (no delay)
+        push_future(&mut futures, operation(first_peer));
 
-        // Create futures with staggered delays
-        for (idx, peer) in peers.into_iter().enumerate() {
-            let delay = Duration::from_millis(staggered_delay_ms * idx as u64);
-            let fut = operation(peer);
+        let mut last_error = NodeError::UnexpectedMessage;
+        let mut next_delay = Box::pin(sleep(staggered_delay * peer_index.load(Ordering::SeqCst)));
 
-            let delayed_fut = async move {
-                if delay.as_millis() > 0 {
-                    sleep(delay).await;
+        // Phase 1: Race between futures completion and delays (while alternatives might exist)
+        loop {
+            // Exit condition: no futures running and can't start any more
+            if futures.is_empty() {
+                if let Some(peer) = self.in_flight_tracker.pop_alternative_peer(key).await {
```

Comment:
Should we check here in case `first_peer` was already registered as an alternative peer, or can't that happen anyway?